dashboard/test/models/sections/section_test.rb

```diff
@@ -71,6 +71,78 @@ class SectionTest < ActiveSupport::TestCase
     end
   end
 
+  test 'changing sharing_disabled to true updates user settings' do
+    student = create :student, sharing_disabled: false
+    section = Section.create @default_attrs
+    assert_creates(Follower) do
+      section.add_student student
+    end
+    section.sharing_disabled = true
+
+    # Student update happens on before save
+    section.save
+    student.reload
+    assert student.sharing_disabled?
+  end
+
+  test 'changing sharing_disabled to false maintains user settings' do
+    student = create :student, sharing_disabled: false
+    section = Section.create @default_attrs
+    assert_creates(Follower) do
+      section.add_student student
+    end
+
+    # Student update happens on before save
+    section.save
+    student.reload
+    assert !student.sharing_disabled?
+  end
+
+  test 'adding student updates their share setting when section share is disabled' do
+    section = Section.create @default_attrs
+    section.sharing_disabled = true
+
+    student = create :student, sharing_disabled: false
+    assert_creates(Follower) do
+      section.add_student student
+    end
+
+    assert student.sharing_disabled?
+  end
+
+  test 'adding student updates their share setting when section share is enabled' do
+    section = Section.create @default_attrs
+    section.sharing_disabled = false
+
+    student = create :student, sharing_disabled: false
+    assert_creates(Follower) do
+      section.add_student student
+    end
+
+    assert !student.sharing_disabled?
```

Comment:
Again this doesn't seem to be doing anything. The student is created with `sharing_disabled` already `false`.

If you want to assert that it changes, start it true. On line 117:
```ruby
student = create :student, sharing_disabled: true
```

Also (ruby-style) use `refute` instead of `assert !`